src/mapped_nca.cpp

```diff
@@ -206,8 +206,8 @@ const uint8_t *MappedNca::section_data(int section) {
   return _backing_data->_data + section_offset;
 }
 
-std::string MappedNca::read_aes_ctr_aligned(int section, uint64_t data_offset,
-                                            uint64_t len) {
+void MappedNca::read_aes_ctr_aligned(int section, uint64_t data_offset,
```

Comment:
why out parameter here but the caller just returns the string? was this function meant to return some sort of status code (just a guess based on the asserts)?

fwiw, I'd like if they both worked with `uint8_t` pointers.

Reply:
there are multiple callers - one is the wrapper that returns managed data (a `std::string`), but the other caller is the modified delta code that already has a buffer ready and just wants to write into it